geode-core/src/test/java/org/apache/geode/internal/cache/FireAndForgetFunctionOnAllServersDUnitTest.java

```diff
@@ -0,0 +1,164 @@
+    disconnectAllFromDS();
+  }
+
+  @Test
+  public void testFireAndForgetFunctionOnAllServers() {
+
+    // Test case for Executing a fire-and-forget function on all servers as opposed to only
+    // executing on the ones the
+    // client is currently connected to.
+
+    Host host = Host.getHost(0);
+    VM locator = host.getVM(0);
+    VM server1 = host.getVM(1);
+    VM server2 = host.getVM(2);
+    VM client = host.getVM(3);
+
+    final int locatorPort = AvailablePort.getRandomAvailablePort(AvailablePort.SOCKET);
+    final String locatorHost = NetworkUtils.getServerHostName(host);
+
+    // Step 1. Start a locator and one cache server.
+    locator.invoke("Start Locator", () -> startLocator(locatorHost, locatorPort, ""));
+
+    String locString = getLocatorString(host, locatorPort);
+
+    // Step 2. Start a server and create a replicated region "R1".
+    server1.invoke("Start BridgeServer",
+        () -> startBridgeServer(new String[] {"R1"}, locString, new String[] {"R1"}));
+
+    // Step 3. Create a client cache with pool mentioning locator.
+    client.invoke("create client cache and pool mentioning locator", () -> {
+      ClientCacheFactory ccf = new ClientCacheFactory();
+      ccf.addPoolLocator(locatorHost, locatorPort);
+      ClientCache cache = ccf.create();
+      Pool pool1 = PoolManager.createFactory().addLocator(locatorHost, locatorPort)
+          .setServerGroup("R1").create("R1");
+
+      Region region1 = cache.createClientRegionFactory(ClientRegionShortcut.PROXY).setPoolName("R1")
+          .create("R1");
+
+      // Step 4. Execute the function to put DistributedMemberID into above created replicated
+      // region.
+      Function function =
+          new TestFunction(false, TestFunction.TEST_FUNCTION_FIREANDFORGET_ONALL_SERVERS);
+      FunctionService.registerFunction(function);
+
+      String regionName = "R1";
+      Execution dataSet = FunctionService.onServers(pool1);
+      dataSet.withArgs(regionName).execute(function);
+
+      // Using Awatility, if the condition is not met during the timeout, a
+      // ConditionTimeoutException will be thrown. This makes analyzing the failure much simpler
+      await().atMost(60, SECONDS).until(() -> {
+        return (region1.keySetOnServer().size() == 1);
+      });
+
+      // Step 5. Assert for the region keyset size with 1.
+      Assert.assertEquals(1, region1.keySetOnServer().size());
+
+      region1.clear();
+
+      // Step 6. Start another server mentioning locator and create a replicated region "R1".
+      server2.invoke("Start BridgeServer",
+          () -> startBridgeServer(new String[] {"R1"}, locString, new String[] {"R1"}));
+
+      // Step 7. Execute the same function to put DistributedMemberID into above created replicated
+      // region.
+      function = new TestFunction(false, TestFunction.TEST_FUNCTION_FIREANDFORGET_ONALL_SERVERS);
+      FunctionService.registerFunction(function);
+
```

Comment:
The test did not verify the pool.getAllServices() has made difference. 

You should add assertEquals(1, pool1.getCurrentServers().size()) at this line.